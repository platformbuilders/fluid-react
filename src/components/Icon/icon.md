
# Icon 

Icons is based in [Material UI](https://material-ui.com/pt/components/material-icons/)

### Source Code
<!-- SOURCE -->

<!-- PROPS -->

<!-- STORY HIDE START -->

content here won't be shown in stories

<!-- STORY HIDE END -->

