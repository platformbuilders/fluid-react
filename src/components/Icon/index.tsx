import React from 'react';
import { Animated, View } from 'react-native';
import MaterialIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import Touchable from '../Touchable';
import * as Icons from '~/assets/svg';

type Props = {
  name: string;
  touchable?: boolean;
  size?: number;
  accessibility?: string;
  color?: string;
  style?: any;
  onPress?(): void;
  borderColor?: string;
  backgroundColor?: string;
  hitSlop?: {
    top?: number;
    bottom?: number;
    left?: number;
    right?: number;
  };
};

export const Icon: React.FC<Props> = ({
  name,
  accessibility,
  touchable = true,
  size = 20,
  color = undefined,
  style = [{}],
  onPress = (): void => {},
  borderColor = '',
  backgroundColor = '',
  ...rest
}) => {
  // eslint-disable-next-line prettier/prettier
  const iconName = name?.charAt(0).toUpperCase() + name?.slice(1);
  const Svg = Icons[`Icon${iconName}`];

  return (
    <Animated.View style={style}>
      <Touchable
        accessibility={accessibility || iconName}
        touchable={!touchable}
        onPress={onPress}
        {...rest}
      >
        <View>
          {Svg ? (
            <Svg
              width={size}
              height={size}
              color={color}
              borderColor={borderColor}
              backgroundColor={backgroundColor}
            />
          ) : (
            <MaterialIcons name={name} color={color} size={size} />
          )}
        </View>
      </Touchable>
    </Animated.View>
  );
};

export default Icon;
