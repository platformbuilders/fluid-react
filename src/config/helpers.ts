import { DefaultTheme } from 'styled-components';
import { animations, colors, radius, spacing } from '../theme';

export const theme: DefaultTheme = {
  ...colors,
  ...animations,
  ...radius,
  ...spacing,
};
