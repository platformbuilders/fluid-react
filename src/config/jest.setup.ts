import { configure } from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16';

configure({ adapter: new EnzymeAdapter() });

// Mocks
jest.mock('@platformbuilders/helpers', () => ({
  getTheme: jest.fn(),
}));
