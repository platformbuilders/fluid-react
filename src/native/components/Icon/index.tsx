import React, { FC } from 'react';
import { Animated, View } from 'react-native';
import MaterialIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import { IconType } from '../../types';
import Touchable from '../Touchable';
import * as Icons from '../../assets/svg';

export const Icon: FC<IconType> = ({
  id,
  accessibility,
  accessibilityLabel,
  testID,
  name = '',
  touchable = true,
  size = 20,
  color = undefined,
  style = [{}],
  onPress = (): void => {},
  borderColor = '',
  backgroundColor = '',
  ProjectIcons,
  ...rest
}) => {
  const iconName = name?.charAt(0).toUpperCase() + name?.slice(1);
  const Svg = ProjectIcons
    ? ProjectIcons[`Icon${iconName}`]
    : Icons[`Icon${iconName}`];

  return (
    <Animated.View style={style}>
      <Touchable
        id={id}
        accessibility={accessibility || iconName}
        accessibilityLabel={accessibilityLabel || accessibility}
        testID={testID || id}
        disabled={!touchable}
        onPress={onPress}
        {...rest}
      >
        <View>
          {Svg ? (
            <Svg
              width={size}
              height={size}
              color={color}
              borderColor={borderColor}
              backgroundColor={backgroundColor}
            />
          ) : (
            <MaterialIcons name={name} color={color} size={size} />
          )}
        </View>
      </Touchable>
    </Animated.View>
  );
};

export default Icon;
