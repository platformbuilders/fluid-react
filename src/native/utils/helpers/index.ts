export * from '@platformbuilders/helpers';
export * from './generateHaptic';
