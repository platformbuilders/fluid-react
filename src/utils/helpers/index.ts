export * from './haptic';
export { default as getTheme } from './getTheme';
export { default as ifStyle } from './ifStyle';
export { default as setStyleIOS } from './setStyleIOS';
export { default as switchStyle } from './switchStyle';
export { isIOS } from './isIOS';
