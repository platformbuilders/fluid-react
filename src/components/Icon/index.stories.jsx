import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobs, text } from '@storybook/addon-knobs';
import markdown from './icon.md';
import Icon from './index';

storiesOf('Icon', module)
  .addDecorator(withKnobs)
  .addParameters({
    readme: {
      content: markdown,
    },
    viewport: {
      defaultViewport: '',
    },
  })
  .add('Example', () => <Icon name={text('name', 'add_circle')} />);
