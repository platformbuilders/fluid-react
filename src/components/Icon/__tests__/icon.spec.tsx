import { shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';

import Icon, { Props } from '..';
import { theme } from '../../../config/helpers';

const defaultProps: Props = {
  name: 'any_name',
};

describe('Component: Icon', () => {
  test('snapshots with default props', () => {
    const component = shallow(
      <ThemeProvider theme={theme}>
        <Icon {...defaultProps} />
      </ThemeProvider>,
    );

    expect(component).toMatchSnapshot();
  });

  test('snapshots with other props', () => {
    const component = shallow(
      <ThemeProvider theme={theme}>
        <Icon {...defaultProps} size="large" color="primary" />
      </ThemeProvider>,
    );

    expect(component).toMatchSnapshot();
  });

  test('check default props', () => {
    // when
    const component = shallow(<Icon {...defaultProps} />);

    // then
    const wrapper = component.props();
    const element = component.children().props();

    expect(wrapper.color).toEqual('inherit');
    expect(wrapper.edge).toEqual('start');

    expect(element.color).toEqual('inherit');
    expect(element.fontSize).toEqual('default');
    expect(element.children).toEqual(defaultProps.name);
  });

  test('check props', () => {
    // should
    const sizeMock = 'small';
    const colorMock = 'error';

    // when
    const component = shallow(
      <Icon {...defaultProps} size={sizeMock} color={colorMock} />,
    );

    // then
    const wrapper = component.props();
    const element = component.children().props();

    expect(wrapper.color).toEqual('inherit');
    expect(wrapper.edge).toEqual('start');

    expect(element.color).toEqual(colorMock);
    expect(element.fontSize).toEqual(sizeMock);
    expect(element.children).toEqual(defaultProps.name);
  });
});
