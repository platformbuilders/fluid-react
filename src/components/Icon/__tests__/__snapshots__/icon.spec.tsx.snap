// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Icon snapshots with default props 1`] = `
<ContextProvider
  value={Object {}}
>
  <Icon
    name="any_name"
  />
</ContextProvider>
`;

exports[`Component: Icon snapshots with other props 1`] = `
<ContextProvider
  value={Object {}}
>
  <Icon
    color="primary"
    name="any_name"
    size="large"
  />
</ContextProvider>
`;
